### PR TITLE
ENH/API: implement discussion in and closes #1954.

### LINKED ISSUE #1954: Consolidating list of new functions
reiterating some points @dengemann made in #1703. there was a discussion about a poliferation of functions in dev0.9. some may redundant, e.g. `set_channel_positions` and the Montage workflow: `read_montage` and `apply_montage`. I think we should try to hash out what we need and don't. I think before the release, we should definitively clear these before we have a deprecation nightmare in the future. If we want to keep the particular function even though it is very similar to another one, we should make it private. These are the things that come to mind. 

Montages
Digitizer Data
Referencing

### PR DESCRIPTION
**Pull request by dengemann**
Cleans up new sensor related functions.

### CODE REVIEW

File: doc/source/whats_new.rst [hunk 1]
```diff
@@ -41,7 +41,7 @@ Changelog
 
    - Add reading and writing support for digitizer data, and function for adding dig points to info by `Teon Brooks`_
 
-   - Add methods ``get_channel_positions``, ``set_channel_positions`` for setting channel positions,
+   - Add methods ``_get_channel_positions``, ``_set_channel_positions`` for setting channel positions,
```
Comment (teonlamont) [inline doc/source/whats_new.rst]: I thought that we typically didn't add private functions to the whats_new

Comment (dengemann) [inline doc/source/whats_new.rst]: :) mass replace

**Comment by agramfort** [comment]
why did you close?

**Comment by dengemann** [comment]
@agramfort because @teonlamont cherry-picked it, as we had some overlapping commit.

**Comment by agramfort** [comment]
Ok do what's best

**Pull request closed without merging**